### DISCUSSION TITLE
How to update user's profile image beyond first login

### DISCUSSION THREAD

**Discussion by karl-barbour** [nextauthjs/next-auth#1414] [Help] [answered]
**Your question**
I'm successfully storing a user's profile image (using the standard Discord provider), but it seems after the user changes their profile image it never updates in the database. How would one go about caching a new image URL on subsequent sign-in?

**Reproduction**
1. Sign in with Discord provider
2. Change avatar
3. Sign out
4. Sign in
5. Old avatar remains forever

**Feedback**
*Documentation refers to searching through [online documentation](https://next-auth.js.org), code comments and issue history. The example project refers to [next-auth-example](https://github.com/iaincollins/next-auth-example).*

* [ ] Found the documentation helpful
* [ ] Found documentation but was incomplete
* [x] Could not find relevant documentation
* [ ] Found the example project helpful
* [ ] Did not find the example project helpful

**Comment by StefanSelfTaught** [comment, marked as answer]
You could update the image in the `signIn` callback.

**Reply by balazsorban44** [reply]
Would do the same! Check out the docs: https://next-auth.js.org/configuration/callbacks#sign-in-callback

**Reply by briskxyz** [reply]
How should I do that, exactly?  I have a mongodb which is mentioned with a single `database: process.env.DATABASE_URL` in `[...nextAuth].js`, and I finally achieved that I can change the current session user's image, but when I refresh the page, it's gone, because it's only in that specific session. Google provider, users stored in the `users` collection in mongodb with that one single reference to the db I mentioned above. Looking forward for your advices. Cheers! 🍻

**Reply by nimeshmaharjan1** [reply]
but usually a app will have something like a profile page or modal updating the image in the database wont change the image returned by the useSession hook

**Reply by goustaff** [reply]
Hello @balazsorban44 
I'm facing the same issue.

Should I open a new discussion, or could we continue here?

I would be glad to know what you mean by "update the image in the `signIn` callback" (for a Discord provider)?
I don't see this use case in the documentation page you linked. I also regret that it is not automated directly by _Next Auth_. :/

**Comment by kfbfarley** [comment]
I did something like this:

```
async createUser(user) {
      if(!user.image){
        const image = 'https://www.gravatar.com/avatar/'+md5(user.email);
        await prisma.user.update({
          where: { id: Number(user.id) },
          data: { image },
        })
      }
    }
```

If you log in with a provider that doesn't provide a user image then it will use gravatar.

I don't really know if it's the best way to do it, but for me, it works.

**Comment by ziedHamdi** [comment]
I'm not the one who saved the image the first time, why should I be the one who has to update it? 
I think this is a bug, we could work around it by updating the image our selves, but it will be a workaround not a solution I think.

**Comment by Janhouse** [comment]
Some solution would be nice. When using database adapter, `getServerSession(authOptions)` returns old data from the database, which never gets updated.
After each login there could be new data (email, groups, other scopes) so it makes no sense that the data is written only once.

And it is not possible to access new session values because data comes from the database only.

**Reply by reinaldo-vombo** [reply]
I had a similar problem where when the user updated their data it was reflected in the database but it did not update in the application. I solved this by forcing a signin when it was successful

    ```
  const result = await UpdateAccount(formData)
      if (result?.error) {
         toast(result?.error)
         ref.current?.reset()
      } else {
         await signIn('credentials', {
            redirect: false,
            email: result.email,
            password: result.password,
         });
```
